peps/pep-0728.rst

```diff
@@ -141,424 +142,422 @@ in TypeScript achieve this:
+subclasses of the TypedDict type may redeclare ``extra_items``.
 
 Because a non-closed TypedDict type implicitly allows non-required extra items
-of value type ``ReadOnly[object]``, its subclass can override the special
-``__extra_items__`` with more specific types.
+of value type ``ReadOnly[object]``, its subclass can override the
+``extra_items`` argument with more specific types.
 
 More details are discussed in the later sections.
 
 Inheritance
 -----------
 
-When the TypedDict type is defined as ``closed=False`` (the default),
-``__extra_items__`` should behave and be inherited the same way a regular key
-would. A regular ``__extra_items__`` key can coexist with the special
-``__extra_items__`` and both should be inherited when subclassing.
-
-We assume that ``closed=True`` whenever ``__extra_items__`` is mentioned for the
-rest of this section.
+``extra_items`` is inherited in a similar way as a regular ``key: value_type``
+item. As with the other keys, the `inheritance rules
+<https://typing.readthedocs.io/en/latest/spec/typeddict.html#inheritance>`__
+and `read-only items inheritance rules
+<https://typing.readthedocs.io/en/latest/spec/typeddict.html#id3>`__ apply.
 
-``__extra_items__`` is inherited the same way as a regular ``key: value_type``
-item. As with the other keys, the same rules from
-`the typing spec <https://typing.readthedocs.io/en/latest/spec/typeddict.html#inheritance>`__
-and :pep:`PEP 705 <705#inheritance>` apply. We interpret the existing rules in the
-context of ``__extra_items__``.
-
-We need to reinterpret the following rule to define how ``__extra_items__``
-interacts with it:
+We need to reinterpret these rules to define how ``extra_items`` interacts with
+them.
 
     * Changing a field type of a parent TypedDict class in a subclass is not allowed.
 
-First, it is not allowed to change the value type of ``__extra_items__`` in a subclass
+First, it is not allowed to change the value of ``extra_items`` in a subclass
 unless it is declared to be ``ReadOnly`` in the superclass::
 
-    class Parent(TypedDict, closed=True):
-        __extra_items__: int | None
+    class Parent(TypedDict, extra_items=int | None):
+        pass
     
-    class Child(Parent, closed=True):
-        __extra_items__: int  # Not OK. Like any other TypedDict item, __extra_items__'s type cannot be changed
+    class Child(Parent, extra_items=int): # Not OK. Like any other TypedDict item, extra_items's type cannot be changed
 
-Second, ``__extra_items__: T`` effectively defines the value type of any unnamed
+Second, ``extra_items=T`` effectively defines the value type of any unnamed
 items accepted to the TypedDict and marks them as non-required. Thus, the above
 restriction applies to any additional items defined in a subclass. For each item
 added in a subclass, all of the following conditions should apply:
 
-- If ``__extra_items__`` is read-only
+.. _inheritance-read-only:
```

Comment:
```suggestion
.. _pep728-inheritance-read-only:
```
I believe these are global in the peps project, so we should namespace them for clarity. (Also need to change the place where it's linked from.)